File: teamcity-vsix-gallery-server/src/main/teamcity/vsix/index/VsixPackageStructureAnalyser.java
package teamcity.vsix.index;

import com.intellij.openapi.diagnostic.Logger;
import com.intellij.openapi.util.text.StringUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.joda.time.DateTime;

import java.util.LinkedHashMap;
import java.util.Map;

import static teamcity.vsix.index.PackageAttributes.*;

public class VsixPackageStructureAnalyser {
    private final Map<String, String> manifestItems = new LinkedHashMap<String, String>();
    private final DateTime finishDate;

    static Logger LOG = Logger.getInstance("teamcity.vsix");

    @NotNull
    public Map<String, String> getItems() {
        return manifestItems;
    }

    public VsixPackageStructureAnalyser(DateTime finishDate) {
        this.finishDate = finishDate;
    }

    public void analyzeVsixManifest(@NotNull VsixManifestContent manifest) {
        addItem(ID, manifest.getId());
        addItem(VERSION, manifest.getVersion());
        addItem(TITLE, manifest.getDisplayName());
        addItem(SUMMARY, manifest.getDescription());
        addItem(LAST_UPDATED, finishDate.toString());
        addItem(PUBLISHER, manifest.getPublisher());
    }

    private void addItem(@NotNull final String key, @Nullable final String value) {
        if (!StringUtil.isEmptyOrSpaces(value)) {
            manifestItems.put(key, value);
        }
    }
}

File: teamcity-vsix-gallery-server/src/main/teamcity/vsix/index/VsixPackageStructureAnalyser.kt
package teamcity.vsix.index

import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.util.text.StringUtil
import org.joda.time.DateTime

import java.util.LinkedHashMap


public class VsixPackageStructureAnalyser(private val finishDate: DateTime) {
    private val manifestItems = LinkedHashMap<String, String>()

    public fun getItems(): MutableMap<String, String> {
        return manifestItems
    }

    public fun analyzeVsixManifest(manifest: VsixManifestContent) {
        addItem(ID, manifest.id)
        addItem(VERSION, manifest.version)
        addItem(TITLE, manifest.displayName)
        addItem(SUMMARY, manifest.description)
        addItem(LAST_UPDATED, finishDate.toString())
        addItem(PUBLISHER, manifest.publisher)
    }

    private fun addItem(key: String, value: String) {
        if (!StringUtil.isEmptyOrSpaces(value)) {
            manifestItems.put(key, value)
        }
    }
}
